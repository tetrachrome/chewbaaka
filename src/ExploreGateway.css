/**
 * ExploreGateway.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 02, 2020
 */

/* .ui.container { */
.outerContainer {
  /* height: 800px; */
  /* margin: 20px 0px; */
  padding: 100px auto;
  background-color: rgba(239, 239, 239, 1.0);
  /* background-color: black; */
}
