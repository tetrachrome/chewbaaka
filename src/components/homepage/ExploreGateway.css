/**
 * ExploreGateway.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 02, 2020
 */
