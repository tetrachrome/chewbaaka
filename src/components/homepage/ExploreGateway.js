/**
 * ExploreGateway.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 02, 2020
 */

import React from 'react';

import "semantic-ui-css/semantic.min.css";

import ExploreGatewayIntro from './ExploreGatewayIntro'
import ExploreGatewayGrid from './ExploreGatewayGrid'

function ExploreGateway() {
  return (
    <div id="ExploreGateway" className="outerContainer" data-testid="ExploreGatewayComponentTestId">
      <ExploreGatewayIntro/>
      <ExploreGatewayGrid/>
      <br/>
    </div>
  )
}

export default ExploreGateway;
