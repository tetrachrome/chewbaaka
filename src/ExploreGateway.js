/**
 * ExploreGateway.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 02, 2020
 */

import React from 'react';

import "semantic-ui-css/semantic.min.css";

import { Container, Segment } from 'semantic-ui-react'

import ExploreGatewayGrid from './ExploreGatewayGrid'

import './ExploreGateway.css';

function ExploreGateway() {
  return (
    <div className="outerContainer">
      <ExploreGatewayGrid/>
      {/* Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa strong. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede link mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. */}
    </div>
  )
}

export default ExploreGateway;
