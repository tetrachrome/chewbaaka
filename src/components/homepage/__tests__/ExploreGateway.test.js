/**
 * ExploreGateway.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 04, 2020
 * Updated  : Aug 06, 2020
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import ExploreGateway from '../ExploreGateway'

import RouterWrapped from '../../../testing/RouterWrapped'

test('renders ExploreGateway component', () => {
  render(
    RouterWrapped(
      <ExploreGateway />
    )
  );

  const expectedNumChildrenComponent = 2;

  // Tests component has the expected number of children components present.
  const mainComponent = screen.getByTestId("ExploreGatewayComponentTestId");
  expect(mainComponent.children.length).toBeGreaterThanOrEqual(expectedNumChildrenComponent);

  // Tests explore gateway intro component to be present.
  const introComponent = screen.getByTestId("ExploreGatewayIntroTestId");
  expect(introComponent).toBeInTheDocument();

  // Tests explore gateway grid component to be present.
  const gridComponent = screen.getByTestId("ExploreGatewayGridTestId");
  expect(gridComponent).toBeInTheDocument();
});

test('ExploreGateway component snapshot', () => {
  const tree = renderer
    .create(
      RouterWrapped(
        <ExploreGateway />
      )
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
