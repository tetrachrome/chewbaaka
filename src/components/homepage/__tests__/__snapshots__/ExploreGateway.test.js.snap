// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreGateway component snapshot 1`] = `
<div
  data-testid="ExploreGatewayComponentTestId"
  id="Explore"
>
  <div
    className="ExploreGatewayIntro"
    data-testid="ExploreGatewayIntroTestId"
  >
    <br />
    <div
      className="ui container"
      width={8}
    >
      <div
        className="ui huge center aligned header"
      >
        Explore CCF's Cheetah Museum in Namibia and learn everything you need to know about cheetahs, all at the comfort of your home.
      </div>
    </div>
    <br />
    <br />
    <div
      className="ui container two column grid centerGrid"
    >
      <div
        className="row"
      >
        <div
          className="eight wide column"
        >
          <div
            className="centerGridLeftColumnStyle"
          >
            <p>
              "CCF is Changing the World to Save the Cheetah" - Dr. Laurie Marker
            </p>
          </div>
        </div>
        <div
          className="seven wide column"
        >
          <div
            className="ui inverted segment"
          >
            <p
              className="JoinTheRace"
            >
              Join the race to 
              <a
                href="https://twitter.com/search?q=%23SaveTheCheetah"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  className="SafeTheCheetah"
                >
                  #SafeTheCheetah
                </span>
              </a>
            </p>
            <p>
              Cheetahs can't win without us.
            </p>
            <a
              className="ui basic inverted button"
              href="https://cheetah.org/get-involved/ways-to-give/"
              onClick={[Function]}
              role="button"
            >
              Get Involved
            </a>
          </div>
        </div>
      </div>
    </div>
    <br />
  </div>
  <div
    className="ui container stackable center aligned four column grid"
    data-testid="ExploreGatewayGridTestId"
  >
    <div
      className="column ExploreGatewayGridItemColumn"
    >
      <div
        className="ExploreGatewayGridItemImageWrapper"
      >
        <a
          className="ExploreGatewayGridItemImageAnchor"
          data-testid="ExploreGatewayGridItemImageAnchorTestId"
          href="history"
        >
          <img
            className="ui medium image ExploreGatewayGridItemImage"
            data-testid="ExploreGatewayGridItemImageTestId"
            src="explore_gateway_grid_item_img_01.jpg"
          />
        </a>
      </div>
      <h2
        className="ui left aligned header ExploreGatewayGridItemHeader"
      >
        History
      </h2>
    </div>
    <div
      className="column ExploreGatewayGridItemColumn"
    >
      <div
        className="ExploreGatewayGridItemImageWrapper"
      >
        <a
          className="ExploreGatewayGridItemImageAnchor"
          data-testid="ExploreGatewayGridItemImageAnchorTestId"
          href="biology"
        >
          <img
            className="ui medium image ExploreGatewayGridItemImage"
            data-testid="ExploreGatewayGridItemImageTestId"
            src="explore_gateway_grid_item_img_02.jpg"
          />
        </a>
      </div>
      <h2
        className="ui left aligned header ExploreGatewayGridItemHeader"
      >
        Biology
      </h2>
    </div>
    <div
      className="column ExploreGatewayGridItemColumn"
    >
      <div
        className="ExploreGatewayGridItemImageWrapper"
      >
        <a
          className="ExploreGatewayGridItemImageAnchor"
          data-testid="ExploreGatewayGridItemImageAnchorTestId"
          href="ecology"
        >
          <img
            className="ui medium image ExploreGatewayGridItemImage"
            data-testid="ExploreGatewayGridItemImageTestId"
            src="explore_gateway_grid_item_img_03.jpg"
          />
        </a>
      </div>
      <h2
        className="ui left aligned header ExploreGatewayGridItemHeader"
      >
        Ecology
      </h2>
    </div>
    <div
      className="column ExploreGatewayGridItemColumn"
    >
      <div
        className="ExploreGatewayGridItemImageWrapper"
      >
        <a
          className="ExploreGatewayGridItemImageAnchor"
          data-testid="ExploreGatewayGridItemImageAnchorTestId"
          href="future"
        >
          <img
            className="ui medium image ExploreGatewayGridItemImage"
            data-testid="ExploreGatewayGridItemImageTestId"
            src="explore_gateway_grid_item_img_04.jpg"
          />
        </a>
      </div>
      <h2
        className="ui left aligned header ExploreGatewayGridItemHeader"
      >
        Future
      </h2>
    </div>
  </div>
  <br />
</div>
`;
